packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -556,7 +556,7 @@ function voidFunctionArguments(
             // Unwrap 'Array<MaybeVoidFunction>' to 'MaybeVoidFunction',
             // so that we'll handle it in the same way as a non-rest
             // 'param: MaybeVoidFunction'
-            type = getTypeArguments(type, checker)[0];
+            type = checker.getTypeArguments(type)[0];
```

Comment:
The `getTypeArguments` wrapper isn't necessary anymore now that we don't support TS 3.7. Removed in https://github.com/typescript-eslint/typescript-eslint/pull/7834/files#diff-565da39ca1ac1312483692e44ec0f5b9cef281d6525f2ac0978b5cd2bb71a828.

Reply:
Actually, _not_ removed, as that'd be a breaking change. Just deprecated.